cscs-checks/apps/greasy/greasy_check.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import itertools
+import os
+import fnmatch
+from datetime import datetime
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+from reframe.core.launchers.registry import getlauncher
+
+
+def to_seconds(str):
+    return (datetime.strptime(str, '%H:%M:%S') -
+            datetime.strptime('00:00:00', '%H:%M:%S')).total_seconds()
+
+
+@rfm.required_version('>=2.19')
+@rfm.parameterized_test(
+    ['serial',     'gpu', 24, 12, 1, 1],
+    ['serial',     'mc',  72, 36, 1, 1],
+    ['openmp',     'gpu', 24,  3, 1, 4],
+    ['openmp',     'mc',  72,  9, 1, 4],
+    ['mpi',        'gpu', 24,  4, 3, 1],
+    ['mpi',        'mc',  72, 12, 3, 1],
+    ['mpi+openmp', 'gpu', 24,  3, 2, 2],
+    ['mpi+openmp', 'mc',  72,  6, 3, 2]
+)
+class GREASYCheck(rfm.RegressionTest):
+    def __init__(self, variant, partition, num_greasy_tasks, nworkes_per_node,
+                 nranks_per_worker, ncpus_per_worker):
+        self.valid_systems = ['daint:' + partition, 'dom:' + partition]
+
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.sourcepath = 'tasks_mpi_openmp.c'
+        self.build_system = 'SingleSource'
+
+        # sleep enough time to distinguish if the files are running in parallel
+        # or not
+        self.sleep_time = 60
+        self.build_system.cflags = ['-DSLEEP_TIME=%d' % self.sleep_time]
+        if variant == 'openmp':
+            self.build_system.cflags += ['-fopenmp']
+        elif variant == 'mpi':
+            self.build_system.cflags += ['-D_MPI']
+        elif variant == 'mpi+openmp':
+            self.build_system.cflags += ['-fopenmp', '-D_MPI']
+
+        self.executable = 'tasks_mpi_openmp.x'
+        self.tasks_file = 'tasks.txt'
+        self.executable_opts = [self.tasks_file]
+        self.greasy_logfile = 'greasy.log'
+        self.keep_files = [self.tasks_file, self.greasy_logfile]
+        nnodes = 2
+        self.use_multithreading = False
+        self.num_greasy_tasks = num_greasy_tasks
+        self.nworkes_per_node = nworkes_per_node
+        self.nranks_per_worker = nranks_per_worker
+        self.num_tasks_per_node = nranks_per_worker * nworkes_per_node
+        self.num_tasks = self.num_tasks_per_node * nnodes
+        self.num_cpus_per_task = ncpus_per_worker
+        self.sanity_patterns = self.eval_sanity()
+        # Reference value is system agnostic
+        refperf = self.sleep_time * num_greasy_tasks / nworkes_per_node / nnodes
```

Comment:
So that PEP8 does not complain:

```suggestion
        refperf = self.sleep_time*num_greasy_tasks / nworkes_per_node / nnodes
```